### PR TITLE
Created meta image for twitter and facebook

### LINKED ISSUE #2682: Dynamically create meta images for threads
Imgix has [a neat text API](https://docs.imgix.com/apis/url/typesetting-endpoint/~text) that lets one generate an image from some text. We should use this and set the `twitter:image` meta tag on thread views to an image of the thread title + maybe the number of messages in the thread.

For example, click this link: http://assets.imgix.net/~text?txt=Test&txtclr=4B5E75

### PR DESCRIPTION
**Pull request by alejandronanez**
**Status**
- [ ] WIP
- [X] Ready for review
- [ ] Needs testing

**Deploy after merge (delete what needn't be deployed)**
- hyperion (frontend)

**Run database migrations (delete if no migration was added)**
NO

**Release notes for users (delete if codebase-only change)**
-

Hello! This is still a WIP in order to fix #2682,  as you can see, in `src/components/head/index.js` I'm calling `generateHeaderImageFromImgix` with this param `https://spectrum.chat/img/apple-icon-144x144-precomposed.png`, the thing is that we need that same image but served from Imgix in order to apply all the necessary _transformations_.

- Where can I find such image?

Thanks!

## Update

Here's how the image will look like

cover-violet

### CODE REVIEW

File: src/components/head/index.js [hunk 1]
```diff
@@ -17,22 +24,8 @@ export default ({ title, description, image, children }: Props) => {
       <meta name="og:description" content={description} />
       <meta name="twitter:title" content={title} />
       <meta name="twitter:description" content={description} />
-      <meta
-        name="twitter:image"
-        content={
```
Comment (alejandronanez) [inline src/components/head/index.js:22]: Also, I cleaned up this a little 🙂

File: src/helpers/images.js [hunk 2]
```diff
@@ -1,24 +1,32 @@
...
+export const FREE_USER_MAX_IMAGE_SIZE_BYTES = 3000000;
+export const PRO_USER_MAX_IMAGE_SIZE_BYTES = 25000000;
+export const FREE_USER_MAX_IMAGE_SIZE_STRING = `${Math.floor(
+  FREE_USER_MAX_IMAGE_SIZE_BYTES / 1000000
+)}mb`;
+export const PRO_USER_MAX_IMAGE_SIZE_STRING = `${Math.floor(
+  PRO_USER_MAX_IMAGE_SIZE_BYTES / 1000000
+)}mb`;
+
+export const generateHeaderImageFromImgix = (
```
Comment (alejandronanez) [inline src/helpers/images.js]: I made this configurable - but I'm not really sure if we're using this function later down the road. Let me know what you think.

**Comment by mxstbr** [comment]
Do you need the logo or the background image? A background image you can find right here: https://spectrum.imgix.net/default_images/cover-violet.png

**Comment by superbryntendo** [comment]
I actually have a TODO to design a more fleshed out version of this currently.

**Comment by superbryntendo** [comment]
image asset: 
image

The ideal future case would look something like this, I think (example: figma):
twitter share card 1

I don't think that's necessarily worth tackling yet though.

**Comment by alejandronanez** [comment]
@uberbryn looks great! Can you please pass me its Imgix url?

**Comment by superbryntendo** [comment]
Good call, lol. http://spectrum.imgix.net/default_images/twitter-share-card.png

**Comment by alejandronanez** [comment]
@uberbryn this is how 144x144 images will look like [image](http://spectrum.imgix.net/default_images/twitter-share-card.png?txtalign=center,middle&txtfit=max&txtcolor=ffffff&txtpad=4&txtsize=12&txtfont=Helvetica,Bold&txt=Suggestion:%20Automatically%20centering%20shape%20based%20on%20their%20centroids&h=144&w=144) - sorry this is terribly small :(

You can update the `txtsize` `h` & `w` params at the end though just to see it bigger.

**Comment by mxstbr** [comment]
This is blocked by the imgix API not allowing us to have multiple pieces of dynamic text. Contacted them to see how we can work around that!

**Comment by mxstbr** [comment]
They responded to me, apparently one can have up to two auto-wrapping text boxes! [Demo](https://sandbox.imgix.com/view?url=https%3A%2F%2Fassets.imgix.net%2Funsplash%2Fdeer.jpg%3Fblend64%3DaHR0cHM6Ly9hc3NldHMuaW1naXgubmV0L350ZXh0P3R4dD10aGlzIGlzIGEgc2Vjb25kIGJveCB3aGljaCBhbHNvIGF1dG8td3JhcHMgd2hlbiB0aGUgdGV4dCBpcyBsb25nIGVub3VnaCBmdWNrIHllcyZ0eHRjbHI9ZmZmJnR4dHNpemU9MjQmdz00MDA%26mark64%3DaHR0cHM6Ly9hc3NldHMuaW1naXgubmV0L350ZXh0P3c9NDAwJnR4dHNpemU9MjQmdHh0Y2xyPWZmZiZ0eHQ9VGhpcyBpcyBhIGZpcnN0IGJveCB3aXRoIGEgbG90IG9mIHRleHQgdGhhdCBzaG91bGQgYXV0by13cmFwIGF0IHRoZSByaWdodCBwbGFjZXMgYW5kIHN0dWZmIGxpa2UgdGhhdA%26markalign%3Dtop%252Cleft%26markpad%3D0%26w%3D640)

Going to play around with this to see if we achieve the result we want!

**Comment by mxstbr** [comment]
Oh my god I'm getting somewhere, look at this:



Not perfect yet, but I'll make it perfect and then push to this branch! Exciting!

**Comment by mxstbr** [comment]
Boom, edge cases covered too! 

Normal title, long community slug:



Extremely long title, normal community slug:



> Note: This isn't perfect yet, ideally it would show an ellipsis, but unfortunately the imgix text API only allows us to automatically add an ellipsis on one-line text 🙄 Good enough for now though

The entire generation of these image URLs is in a `getMetaImage({ title, footer })` method 😍 Super easy!

**Comment by mxstbr** [comment]
Follow-up PR: #3742

**Pull request closed without merging**